internal/network/tcp_conn.go

```diff
@@ -0,0 +1,107 @@
+package network
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+
+	"golang.org/x/net/context"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	frameSizeBytes int = 4
+)
+
+var (
+	byteOrder = binary.BigEndian
+)
+
+var _ Conn = (*tcpConn)(nil)
+
+type tcpConn struct {
+	id         ID
+	underlying net.Conn
+	closed     bool
+}
+
+// DialTCP dials to the given address, assuming a TCP network. The returned Conn
+// is ready to use.
+func DialTCP(addr string) (Conn, error) {
+	conn, err := net.Dial("tcp", addr)
+	if err != nil {
+		return nil, fmt.Errorf("dial tcp: %w", err)
+	}
+	return newTCPConn(conn), nil
+}
+
+func newTCPConn(underlying net.Conn) *tcpConn {
+	id := createID()
+	conn := &tcpConn{
+		id:         id,
+		underlying: underlying,
+	}
+	return conn
+}
+
+func (c *tcpConn) ID() ID {
+	return c.id
+}
+
+func (c *tcpConn) Send(payload []byte) error {
```

Comment:
Maybe add a bit of doc to this?
About why 2 separate components of `framesize` and `payload` are sent?

Reply:
It's a pretty common concept called framing
Instead of writing the packet, you first write the length of the packet (`framesize`) and then the packet `frame`
But yes, I will add some doc